applications/sdp/gpio/CMakeLists.txt

```diff
@@ -9,15 +9,10 @@ cmake_minimum_required(VERSION 3.20.0)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(sdp_gpio)
 
-sdp_assembly_generate("${CMAKE_SOURCE_DIR}/src/hrt/hrt.c")
-sdp_assembly_check("${CMAKE_SOURCE_DIR}/src/hrt/hrt.c")
-sdp_assembly_prepare_install("${CMAKE_SOURCE_DIR}/src/hrt/hrt.c")
+sdp_assembly("src/hrt/hrt.c")
```

Comment:
this shows that this function name is for from ideal as `assembly` has no indication that this file will actually be:
- be stop after compilation proper, to form an assembly file
- an install target will be created
- sources will be added to the `app` library.

This should be improved by a better function name, as well as provided the `app` target as argument will indicate a relation between the call and the `app` target.